### PR TITLE
Ensure cart and order values cannot be less than zero

### PR DESCRIPTION
**Pull request by JPry**
### 🎫 Ticket

QA Issue `#036`

### 🗒️ Description

This ensures that orders and the cart value cannot be less than zero. 

[skip-changelog] *Skipping the changelog as this is fixing a bug in a new feature.*

The cause of this issue was the work done in #3612. In that PR, the coupon calculation was moved to the `get_cart_total()` method out of the `get_cart_subtotal()` method. At that point, I didn't include a check after the dynamic calculations to ensure the total wasn't less than zero.

While fixing that particular issue, I found that the Order creation process also has the wrong values set for new orders. Upon digging into the `Order` class, I found that it is doing its own calculations of totals entirely separate from the cart. So, this PR does the following:

1. Updates the `Abstract_Cart::get_cart_total()` method to ensure we don't end up with a negative total for the cart contents.
1. Updates the `Order::create_from_cart()` method with these changes:
    1. It uses the subtotal and total from the cart instead of manually calculating these values. This allows the cart to handle all of the advanced logic.
    1. It updates the item preparation to use more of the data that is already provided when calling `$cart->get_items_in_cart()`. 
    1. Because technically we're changing the total calculation, I added a filter to allow for changing the order total after additional items have been added, and before the order is created
1. I added unit tests that cover the cart having a greater than zero value, and tests that cover the resulting order having a greater than zero value
1. One of the snapshots had an error in the Fee calculation that I didn't notice previously. The fixes above revealed this error. It should be `$15` (`$10` ticket plus `$5` fee), so the snapshot has now been updated.
1. When testing these changes, I noticed that the Checkout page doesn't adjust to handle an order that costs $0 due to a coupon. For that reason, I updated the Coupons API and the frontend JS to trigger a page reload if needed. The reason for the reload is to ensure the correct page template is displayed after a coupon is applied/removed.

### 🎥 Artifacts 

**Before:**

https://github.com/user-attachments/assets/b8934be9-c231-4fba-9b57-f04b96aa16c8

**After:**

https://github.com/user-attachments/assets/8fe9fea5-b07a-45cf-95b6-5e1c8ea9a69f

### ✔️ Checklist
- [ ] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [x] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [x] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

### CODE REVIEW

**Review by Camwyn — APPROVED** [review]
Assuming that the order total line does not include Fees

**Review by dpanta94 — CHANGES_REQUESTED** [review]

File: tests/order_modifiers_integration/Checkout/__snapshots__/Fees_Test__it_should_render_fees__0.snapshot.html [hunk 1]
```diff
@@ -53,7 +53,7 @@
 					<strong>Total</strong>
 				</td>
 				<td style="padding-left:0;">
-					<div class="tec-tickets-commerce-price-container"><ins><span class="tec-tickets-commerce-price">&#x24;10.00</span></ins></div>				</td>
+					<div class="tec-tickets-commerce-price-container"><ins><span class="tec-tickets-commerce-price">&#x24;15.00</span></ins></div>				</td>
```
[the remark this answers is not in the document]
Comment (JPry) [reply tests/order_modifiers_integration/Checkout/__snapshots__/Fees_Test__it_should_render_fees__0.snapshot.html:56]: Yes, a previous PR set the wrong value and I didn't catch it. This one tripped when I ran the tests, and I looked at it and realized *this* value is correct. So the wrong value got saved before.

**Review by dpanta94 — APPROVED** [review]
